docs/sbom-protocol.md

```diff
@@ -0,0 +1,72 @@
+# SBOM Scanning Protocol
+
+BuildKit supports automatic creation of [SBOMs](https://en.wikipedia.org/wiki/Software_supply_chain)
+for builds, attaching them as [image attestations](./attestation-storage.md).
+    
+To scan the filesystem contents, a user can specify an SBOM generator image.
+When run, this image is passed the rootfs of the build stage as a read-only
+mount, writes its SBOM scan data to a specified directory.
+
+The SBOM generator image is expected to follow the rules of the BuildKit SBOM
+generator protocol, defined in this document.
+
+> **Note**
+>
+> Currently, only SBOMs in the [SPDX](https://spdx.dev) JSON format are
+> supported.
+>
+> These SBOMs will be attached to the final image as an in-toto attestation
+> with the `https://spdx.dev/Document` predicate type.
+
+## Implementations
+
+The following SBOM generator images are available:
+
+- [jedevc/buildkit-syft-scanner](https://github.com/jedevc/buildkit-syft-scanner)
+
+## Parameters
+
+When run, the SBOM generator image will be passed various parameters as
```

Comment:
> image will be passed various parameters

Better to just say "These parameters will be passed to the generator image as environment variables by BuildKit:"